benchmarks/src/main/scala/zio/stm/TReentrantLockBenchmark.scala

```diff
@@ -176,28 +175,29 @@ class TReentrantLockBenchmark {
     }
   }
 
-  @CompilerControl(CompilerControl.Mode.DONT_INLINE)
-  def reentrantLockWrite(): Unit = {
-    reentrantLock.lock()
-    doWork()
-    reentrantLock.unlock()
-  }
-
-  @CompilerControl(CompilerControl.Mode.DONT_INLINE)
-  def reentrantLockRead(): Unit = {
-    reentrantLock.lock()
-    doWork()
-    reentrantLock.unlock()
-  }
-
   @CompilerControl(CompilerControl.Mode.DONT_INLINE)
   def stampedLockWrite(): Unit = {
     val stamp = stampedLock.writeLock()
     doWork()
     stampedLock.unlockWrite(stamp)
-  }
+  }*/
+
+  @CompilerControl(CompilerControl.Mode.DONT_INLINE)
+  def reentrantLockWrite(): Unit =
+    for (_ <- calls) {
+      reentrantLock.lock()
+      doWork()
+      reentrantLock.unlock()
+    }
 
   @CompilerControl(CompilerControl.Mode.DONT_INLINE)
+  def reentrantLockRead(): Unit =
+    for (_ <- calls) {
+      reentrantLock.lock()
+      doWork()
```

Comment:
I would delete `doWork` and `doWorkM`, we don't want to measure the cost of doing any work, even a function call. We want to measure only the lock/unlock overhead.

Reply:
won't this have the effect of significantly reducing contention? the locks will be for such a short time that they'll rarely conflict

Reply:
OK, you're right about that. Let's do `ZIO.yieldTo` for `doWorkM`

For `doWork`, we are going to have to do a sleep, but the question is how long? If we want it to be fair, it would be the same "size" as `doWorkM` (which involves submitting work to a thread pool, and having a thread pick up and execute that work).

Reply:
I did fiddle with this a bit, main issue is even the smallest sleep (1 ms) kills so many iterations that we probably need to run the  benchmark longer to get an accurate measure. I think we probably need to do something crazy like actually update some data structure. Something like the zio "compute pi" exercise - both the readers and writers have to do something to give some level of contention (updating the state or computing in vs out for the readers), but that will still be less than 1 ms, and it's a somewhat legitimate use case; the whole reason for having locks is to prevent concurrent updates in this sort of situation (I'm always suss on benchmarks that don't do anything somewhat useful in the real world).